packages/flutter_test/lib/src/accessibility.dart

```diff
@@ -0,0 +1,340 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:math' as math;
+import 'dart:typed_data';
+import 'dart:ui' as ui;
+
+import 'package:flutter/rendering.dart';
+import 'package:flutter/semantics.dart';
+import 'package:flutter/widgets.dart';
+
+import 'finders.dart';
+import 'widget_tester.dart';
+
+/// The result of evaluating a semantics node by a [AccessibilityGuideline].
+class Evaluation {
+  /// Create a passing evaluation.
+  const Evaluation.pass() : passed = true, reason = null;
+
+  /// Create a failing evaluation, with an optional [reason] explaining the
+  /// result.
+  const Evaluation.fail([this.reason]) : passed = false;
+
+  // private constructor for adding cases together.
+  const Evaluation._(this.passed, this.reason);
+
+  /// Whether the given tree or node passed the policy evaluation.
+  final bool passed;
+
+  /// If [passed] is false, contains the reason for failure.
+  final String reason;
+
+  /// Combines two evaluation results.
+  ///
+  /// The [reason] will be concatenated with a newline, and [passed] will be
+  /// combined with an `&&` operator.
+  Evaluation operator +(Evaluation other) {
+    if (other == null)
+      return this;
+    final StringBuffer buffer = new StringBuffer();
+    if (reason != null)
+      buffer.write(reason);
+    if (other.reason != null)
+      buffer.write(other.reason);
+    return new Evaluation._(passed && other.passed, buffer.isEmpty ? null : buffer.toString());
+  }
+}
+
+/// An accessibility guideline describes a recommendation an application should
+/// meet to be considered accessible.
+abstract class AccessibilityGuideline {
+  /// A const constructor allows subclasses to be const.
+  const AccessibilityGuideline();
+
+  /// Evaluate whether the current state of the `tester` conforms to the rule.
+  FutureOr<Evaluation> evaluate(WidgetTester tester);
+
+  /// A description of the policy restrictions and criteria.
+  String get description;
+}
+
+class _MinimumTapTargetGuideline extends AccessibilityGuideline {
+  const _MinimumTapTargetGuideline(this.size);
+
+  final Size size;
```

Comment:
nit: maybe document that this is the expected minimum size?

Reply:
Done